scripts/sizeSnapshot/create.js

```diff
@@ -48,12 +48,73 @@ async function getWebpackSizes() {
   });
 }
 
+// waiting for String.prototype.matchAll in node 10
+function* matchAll(string, regex) {
+  let match = null;
+  do {
+    match = regex.exec(string);
+    if (match !== null) {
+      yield match;
+    }
+  } while (match !== null);
+}
+
+/**
+ * Inverse to `pretty-bytes`
+ *
+ * @param {string} n
+ * @param {'B', 'kB' | 'mB' | 'gB' | 'tB' | 'pB'} unit
+ * @returns {number}
+ */
+function prettyBytesInverse(n, unit) {
+  const metrixPrefix = unit.length < 2 ? '' : unit[0];
+  const metricPrefixes = ['', 'k', 'm', 'g', 't', 'p'];
+  const metrixPrefixIndex = metricPrefixes.indexOf(metrixPrefix);
+  if (metrixPrefixIndex === -1) {
+    throw new TypeError(
+      `unrecognized metric prefix '${metrixPrefix}' in unit '${unit}'. only '${metricPrefixes.join(
+        "', '",
+      )}' are allowed`,
+    );
+  }
+
+  const power = metrixPrefixIndex * 3;
+  return n * 10 ** power;
+}
+
+/**
+ * parses output from next build to size snapshot format
+ * @returns {[string, { gzip: number, files: number, packages: number }][]}
+ */
+async function getNextPagesSize() {
```

Comment:
Do we need to keep track of all the documentation pages? If I could trade all the documentation pages with all the component bundle size, I would :).

Reply:
> Do we need to keep track of all the documentation pages?

It helps spotting ES modules issues

> If I could trade all the documentation pages with all the components bundle size, I would :).

You don't need to trade them. Just add them to the tracking.

Reply:
> It helps spotting ES modules issues

Ok, I was assuming that tracking some key pages, like the home and the button, text field pages would be enough to fix systematic problems.

My question had the following background motivation: tracking all the documentation pages opens the door to optimize specific pages in particular. I don't think it would yield the best return on our time. So I was wondering if we could not collect the data to prevent this in the first place.

However, the counter-argument to it is that we might spot some pages that have an abnormal bundle size, most likely related to third-party libraries we bundle, from the demos. Given people copy & paste the demo, our users will be impacted too.

Reply:
> tracking all the documentation pages opens the door to optimize specific pages in particular. I don't think it would yield the best return on our time. So I was wondering if we could not collect the data to prevent this in the first place.

Yeah we need to moderate ourselves if it comes to that. But specifically #16046  will target various pages. Only tracking pages with demos might make more sense but then other things might fall through the cracks. 

I'd rather have the data available and then limit the view if it becomes to noisy instead of limiting the data.

Reply:
👍